spec/features/publish/courses/editing_course_information_guidance_spec.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+feature 'Guidance components on course edit page', { can_edit_current_and_next_cycles: false } do
+  describe 'university providers' do
+    context 'when provider is excluded via config' do
+      scenario 'Guidance is not shown' do
+        given_a_provider_which_does_not_show_guidance
+        and_the_provider_has_a_course
+        and_i_am_authenticated_as_a_provider_user
+        when_i_go_to_edit_a_course
+        i_do_not_see_the_guidance_text
+      end
+
+      def given_a_provider_which_does_not_show_guidance
+        @provider = create(:provider, :university, provider_code: 'B31')
+      end
+    end
+
+    context 'when provider is not excluded via config' do
+      scenario 'Guidance is shown' do
+        given_a_provider_which_does_show_guidance
+        and_the_provider_has_a_course
+        and_i_am_authenticated_as_a_provider_user
+        when_i_go_to_edit_a_course
+        i_do_see_the_guidance_text
+      end
+
+      def given_a_provider_which_does_show_guidance
+        @provider = create(:provider, :university, provider_code: 'XXX')
+      end
+    end
+  end
+
+  describe 'scitt courses' do
+    context 'when provider is excluded via config' do
+      scenario 'Guidance is not shown' do
+        given_a_provider_which_does_not_show_guidance
+        and_the_provider_has_a_course
+        and_i_am_authenticated_as_a_provider_user
+        when_i_go_to_edit_a_course
+        i_do_not_see_the_guidance_text
+      end
+
+      def given_a_provider_which_does_not_show_guidance
+        @provider = create(:provider, :scitt, provider_code: 'E65')
+      end
+    end
+
+    context 'when provider is not excluded via config' do
+      scenario 'Guidance is shown' do
+        given_a_provider_which_does_show_guidance
+        and_the_provider_has_a_course
+        and_i_am_authenticated_as_a_provider_user
```

Comment:
Do you think it's worth putting the authentication in a before block since it's shared between all of these tests? 

Like: https://github.com/DFE-Digital/publish-teacher-training/blob/4057005e39d40301d9c68a37073898a451468bfb/spec/features/publish/managing_schools_spec.rb#L6-L10

Reply:
I was having trouble changing the authenticated provider in the spec so I decided to create the provider before authenticating on each spec because of how the authenticating method is implemented. I didn't want to make changes to that.